### PR TITLE
PyQt6 and new anki version

### PR DESCRIPTION
**Pull request by ganqqwerty**
Could you please check if the addon still works after moving to PyQt6 and new anki version? 

There were several problems that were causing errors, and I fixed them. However I'm not sure that the hooks are still applying properly. 

Hopefully this PR will be more useful than just creating an issue. I really like the idea of a smart leech control, it's very important in language learning to have smth like that.

### CODE REVIEW

File: src/consts.py [hunk 1]
```diff
@@ -6,13 +6,12 @@
...
 import aqt
 from typing import NewType
-from anki import buildinfo
 from aqt.utils import tr
 from aqt.qt import QT_VERSION_STR
 
 CURRENT_VERSION = '1.3.6'
 
-CURRENT_ANKI_VER = int(buildinfo.version.replace('2.1.', ''))
+CURRENT_ANKI_VER = 66 # hardcoding the anki version to the last 2.1.x
```
Comment (ganqqwerty) [inline src/consts.py:14]: a workaround for not having a problem with parsing the new anki version. Anki changed versioning scheme.

File: src/reviewer.py [hunk 2]
```diff
@@ -197,7 +198,7 @@ def append_hooks(self):
         if CURRENT_ANKI_VER > ANKI_LEGACY_VER and mw.col.v3_scheduler():
             reviewer_did_answer_card.append(self.on_answer_v3)
         else:
-            card_did_leech.append(self.save_leech)
+            hooks.card_did_leech.append(self.save_leech)
```
Comment (ganqqwerty) [inline src/reviewer.py:201]: i'm not sure about this place, please check

File: src/reviewer.py [hunk 3]
```diff
@@ -249,6 +250,8 @@ def remove_hooks(self):
             hooks.card_did_leech.remove(self.save_leech)
         except NameError:
             print(ErrorMsg.ACTION_MANAGER_NOT_DEFINED)
+        except AttributeError:
```
Comment (ganqqwerty) [inline src/reviewer.py:253]: for some reason, sometimes the hooks.card_did_leech does not have anything, in this case we do nothing. Hopefully it won't have any negative effects

File: res/ui/actions_form.py [hunk 4]
```diff
@@ -237,131 +235,131 @@ def setupUi(self, ActionsForm):
...
+        self.frame_9.setFrameShape(QtWidgets.QFrame.Shape.StyledPanel)
+        self.frame_9.setFrameShadow(QtWidgets.QFrame.Shadow.Raised)
         self.frame_9.setObjectName("frame_9")
         self.horizontalLayout_20 = QtWidgets.QHBoxLayout(self.frame_9)
         self.horizontalLayout_20.setObjectName("horizontalLayout_20")
-        self.label_4 = QtWidgets.QLabel(self.frame_9)
+        self.label_4 = QtWidgets.QLabel(parent=self.frame_9)
         self.label_4.setObjectName("label_4")
         self.horizontalLayout_20.addWidget(self.label_4)
         self.queueRatioSlider = TipSlider(self.frame_9)
```
Comment (ganqqwerty) [inline res/ui/actions_form.py:338]: note that here I manually changed the code that pyuic6 outputted. The TipSlider does not have the parent attribute.

**Review by iamjustkoi — APPROVED** [review]
Little concerned about hard-coding the forms and version scheme but definitely wanna try and update those to have a similar functionality to [Study Time Stats](https://github.com/iamjustkoi/StudyTimeStats) in the future, anyway.

Other than that, looks good and will try and test in the main release in a bit here!

Thanks for submitting! ^^

**Comment by iamjustkoi** [comment]
Oh wait that was an approve whoops haha, but it works with some tests and should be good to go!
Thanks again!

**Pull request merged**